### ISSUE TITLE
Error in example

### ISSUE THREAD

**Issue by pmpeter1** [01org/Intel-Pattern-Matching-Technology#40]
byte getAverageSample(byte samples[], unsigned int num, unsigned int pos, unsigned int step)
{
    unsigned int ret;

    /* This is the number of samples that will be used to create each
     * average sample; i.e. all the skipped samples before and after
     * the current sample */
    unsigned int size = step * 2;

    /* Don't do any averaging, if we are at the beginning or end
     * of the sample window */
    if (pos < (step * 3) || pos > (num * 3) - (step * 3)) {
        ret = samples[pos];
    } else {
        ret = 0;
        pos -= (step * 3);

        /* Calculate the sum of 'step' samples before, and after,
         * the current sample */
        for (unsigned int i = 0; i < size; ++i) {
            ret += samples[pos - (3 * i)];    **// ERROR: I think this should be pos + (3*i) since you already decremented pos right before this.**   
        }

        ret /= size;
    }

    return (byte)ret;
}

**Comment by eriknyquist** [comment]
Yeah, I think you're right. Thanks

**Comment by eriknyquist** [comment]
Sorry, haven't had a chance to test/work on this yet, should do in the next week or two

**Comment by eriknyquist** [comment]
You're right, this was an error. Tested out the application with the fix and it still works fine
